src/Middleware/OutputCaching/src/DefaultOutputCachePolicyProvider.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.OutputCaching;
+
+/// <inheritdoc />
+public class DefaultOutputCachePolicyProvider : IOutputCachePolicyProvider
+{
+    private readonly OutputCacheOptions _options;
+
+    /// <summary>
+    /// Creates a new instance of <see cref="DefaultOutputCachePolicyProvider"/>.
+    /// </summary>
+    /// <param name="options">The options configured for the application.</param>
+    public DefaultOutputCachePolicyProvider(IOptions<OutputCacheOptions> options)
+    {
+        ArgumentNullException.ThrowIfNull(options);
+
+        _options = options.Value;
+    }
+
+    /// <inheritdoc />
+    public ValueTask<IOutputCachePolicy?> GetPolicyAsync(string policyName)
+    {
+        ArgumentNullException.ThrowIfNull(policyName);
+
+        var policy = _options.NamedPolicies?[policyName];
```

Comment:
Use `TryGetValue()` to guard against `KeyNotFoundException`.

```suggestion
        if (_options.NamedPolicies?.TryGetValue(policyName, out var policy) is false)
        {
            policy = null;
        }
```

Reply:
The issue is that the `policy` variable might not be initialized when it's used in `ValueTask.FromResult(policy)` if `TryGetValue` returns `false`. When `TryGetValue` returns `false`, `policy` is not assigned any value, which leads to the error. Suggested implementation gives `error CS0165: Use of unassigned local variable 'policy'`.

````csharp


    /// <inheritdoc />
    public ValueTask<IOutputCachePolicy?> GetPolicyAsync(string policyName)
    {
        ArgumentNullException.ThrowIfNull(policyName);

        if (_options.NamedPolicies?.TryGetValue(policyName, out var policy) is false)
        {
            policy = null;
        }

        return ValueTask.FromResult(policy);
    }
````

Reply:
In that case just refactor it to not use `?` and do something like this:

```chsarp
/// <inheritdoc />
public ValueTask<IOutputCachePolicy?> GetPolicyAsync(string policyName)
{
    ArgumentNullException.ThrowIfNull(policyName);

    IOutputCachePolicy? policy = null;

    if (_options.NamedPolicies is not null &&
        _options.NamedPolicies.TryGetValue(policyName, out var value))
    {
        policy = value;
    }

    return ValueTask.FromResult(policy);
}
```

Reply:
Done